### ISSUE TITLE
Implement scrub playback tracking in timeline view

### ISSUE THREAD

**Issue by Jose-Moreno** [pencil2d/pencil#1105]
### Issue Summary

After using the timeline for a bit, I noticed that during playback and when manipulating the playback scrub header it's difficult to track it after it "leaves" the current view.

I feel the scrub should always be visible on the timeline view either during playback, when we use the start / end buttons on the playback controls or when we use the frame / keyframe jumping shortcuts.

This would mean the timeline view should "keep track" of the scrub until it reaches the end of the animation (last keyframe). 

A possible behaviour would be to start following the scrub only when the scrub reaches between 50% ~ 70% of the timeline (or similar to the frame length algorithm) moving the timeline horizontal scroll to match the scrub.
That way there wouldn't be an issue when viewing the first keyframe and tracking up to the last keyframe would be possible as well. 

As a separate case if the user get's to the last maximum frame possible (9999) the timeline should stop tracking when you reach the last portion of the timeline (about ~1% of the current frame length)

**Comment by davidlamhauge** [comment]
Shouldn't this be closed?

**Comment by Jose-Moreno** [comment]
@davidlamhauge I didn't know the PR was merged. I don't have time to test this right now but i'll be free next week, so once I test it I'll close it, apologies for the delay.

**Comment by Jose-Moreno** [comment]
@davidlamhauge Hi David, I tested the latest master, and well so far the red time indicator seems to be getting "followed" only when it gets to the last part of the visible panel and that's fine.

One thing though is that when the playback occurs, if you drag the horizontal scroll bar during playback the timeline does not update and the panel will incorrectly display the frame numbers, and it will updated only after it reaches the final part of the panel.

It was my understanding that this issue had also been fixed on your PR, so I'm not sure why I can't see the effects of that. I'm using Windows 7 Ultimate for my tests by the way. Thanks for your work :slightly_smiling_face:

**Comment by Jose-Moreno** [comment]
Ok I finally tested David's #1257 on the latest dev build. think this is good enough.

Right now the only issue I see is a rare corner case where dragging the horizontal scrollbar for a long animation, and doing so during playback will not properly update the frame dial numbers (unlike when the playback is paused) . The timeline visible region will only update until the playback indicator reaches near the end of the visible portion of the timeline.

That means, even if we move the timeline region during playback, only the current frame that is being read by the playback indicator gets updated, but the regular dial numbers display at the top will not be updated until the playback conditional case is triggered and the region is updated as a whole.

I'll probably open a new issue for that, but it's mostly a rare user experience case, so it is extremely low priority in comparison.

Good work everyone, thanks!
